File: QRCodeScan/QRCodeScan/ViewController.m
//
//  ViewController.m
//  QRCodeScan
//
//  Created by youmy on 16/3/6.
//  Copyright © 2016年 youmy. All rights reserved.
//

#import "ViewController.h"
@interface ViewController ()

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    self.view.backgroundColor = [UIColor whiteColor];
    [self creatScanBtn];
}

/**
 *  创建扫描按钮
 */
- (void)creatScanBtn{
    UIButton * scanBtn = [[UIButton alloc] init];
    [scanBtn setTitle:@"扫描" forState:UIControlStateNormal];
    [scanBtn setTitleColor:[UIColor orangeColor] forState:UIControlStateNormal];
    [scanBtn sizeToFit];
    scanBtn.center = self.view.center;
    [scanBtn addTarget:self action:@selector(scanBtnClick) forControlEvents:UIControlEventTouchUpInside];
    [self.view addSubview:scanBtn];
}

/**
 *  打开扫描界面
 */
- (void)scanBtnClick {
    UIStoryboard * mainStoryBoard = [UIStoryboard storyboardWithName:@"QRCode" bundle:nil];
    UIViewController * vc = mainStoryBoard.instantiateInitialViewController;
    [self presentViewController:vc animated:YES completion:nil];
}

@end

File: QRCodeScanSwiftDemo/ViewController.swift
//
//  ViewController.swift
//  QRCodeScanSwiftDemo
//
//  Created by youmy on 2018/11/29.
//  Copyright © 2018 youmy. All rights reserved.
//

import UIKit

class ViewController: UIViewController {

    override func viewDidLoad() {
        super.viewDidLoad()
        
        let btn = UIButton(frame: CGRect(x: 100, y: 100, width: 100, height: 50))
        btn.setTitle("扫描", for: .normal)
        btn.setTitleColor(.red, for: .normal)
        btn.addTarget(self, action: #selector(scan(sender:)), for: .touchUpInside)
        view.addSubview(btn)
    }

    @objc func scan(sender:UIButton){
        let vc = QRCViewController()
        vc.isBarCode = true
        vc.delegate = self
        navigationController?.pushViewController(vc, animated: true)
    }
}

extension ViewController: QRCViewControllerDelegate {
    func didFinishedSacn(controller: QRCViewController, result: String) {
        print("--->result:\(result)")
    }
}
